holochain-go: init at 0.1.0-alpha

gavinrogers opened the pull request:
Holochain is a platform for creating an unenclosable carrier upon which
highly-scalable dApps and smart contracts can be built. This is the holochain
team's initial implimentation in go. A rust version is under development.

###### Motivation for this change

Add the holochain core to nixpkgs to allow app developers to bootstrap more quickly: allow people to run their own holochain node.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"` (nothing depends on this change)
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after) 
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Commit:
init: holochain-go at 0.1.0-alpha

Holochain is a platform for creating an unenclosable carrier upon which
highly-scalable dApps and smart contracts can be built. This is the holochain
team's initial implimentation in go. A rust version is under development.

Commit:
Cleanup: remove autogenerated line

Agree with @Mic92: "Since this was manually edited, we should not claim it was
auto generated.

Commit:
Cleanup: remove line left over from template

As per @Mic92: can be removed

Commit:
Cleanup: remove v from version and use fetchFromGithub

More housecleaning, also as per @Mic92. He's right, as it says in the manual, to
use fetchFromGithub instead of fetchgit

> Best: Fetches a snapshot archive and you get the rev you want.

Commit:
Fix: Capitalisation of downloadPage

Commit:
Fix: switch expression to use fetchFromGitHub

gavinrogers commented:
@zimbatm How does this look to you?

gavinrogers commented:
@edolstra this is what I was telling you about last week

zimbatm commented on pkgs/servers/holochain-go/default.nix:
```diff
@@ -0,0 +1,30 @@
+    downloadPage = "https://developer.holochain.org/";
+    license = stdenv.lib.licenses.gpl3;
+    maintainers = with stdenv.lib.maintainers; [ gavin ];
+    platforms = stdenv.lib.platforms.all;
```
if `platforms` is not set it will default to `go.meta.platforms` which is probably what you want.

gavinrogers replied:
Ok I'll remove the platforms metadata

zimbatm reviewed (approved):
looking good!

zimbatm commented:
@GrahamcOfBorg holochain-go

xeji commented:
@GrahamcOfBorg build holochain-go

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42937&attempt_id=a7e2293c-512c-4af2-a4b1-9ed4a7478662)

Attempted: holochain-go

<details><summary>Partial log (click to expand)</summary><p>

```
github.com/libp2p/go-libp2p-crypto
golang.org/x/crypto/sha3
github.com/ipfs/go-log/tracer
github.com/multiformats/go-multihash
github.com/ipfs/go-log
github.com/libp2p/go-libp2p-peer
github.com/holochain/holochain-proto/hash
github.com/holochain/holochain-proto/peerqueue
builder for '/nix/store/salvhdi2bzrrm9yd0p0hz4z59650aylq-holochain-go0.1.0-alpha.drv' failed with exit code 8
error: build of '/nix/store/salvhdi2bzrrm9yd0p0hz4z59650aylq-holochain-go0.1.0-alpha.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42937&attempt_id=239d1c14-987f-4fcf-9317-d87918a8dd6c)

Attempted: holochain-go

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/49h8dn5qlrxgcwkaxkkq1971b9334abx-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/bin/test.examples: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/bash"
/nix/store/49h8dn5qlrxgcwkaxkkq1971b9334abx-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/bs.start: interpreter directive changed from "/bin/sh" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/sh"
/nix/store/49h8dn5qlrxgcwkaxkkq1971b9334abx-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/hc.sync: interpreter directive changed from "/bin/sh" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/sh"
/nix/store/49h8dn5qlrxgcwkaxkkq1971b9334abx-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/post.up: interpreter directive changed from "/bin/sh" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/sh"
/nix/store/49h8dn5qlrxgcwkaxkkq1971b9334abx-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/pre.up: interpreter directive changed from "/bin/sh" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/sh"
/nix/store/49h8dn5qlrxgcwkaxkkq1971b9334abx-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/sync: interpreter directive changed from "/bin/sh" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/sh"
/nix/store/49h8dn5qlrxgcwkaxkkq1971b9334abx-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/syncCheckFiles: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/bash"
/nix/store/49h8dn5qlrxgcwkaxkkq1971b9334abx-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/syncStart: interpreter directive changed from "/bin/sh" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/sh"
/nix/store/49h8dn5qlrxgcwkaxkkq1971b9334abx-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/scriptsAccessibleByDockerfiles/entrypoint.sh.addHostUserToContainer: interpreter directive changed from "/bin/sh" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/sh"
/nix/store/fjwzxxr4cb9yy7x9qb1a2cjp396bicys-holochain-go0.1.0-alpha-bin
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42937&attempt_id=198763d1-b504-429f-a0b6-508cfc4493da)

Attempted: holochain-go

<details><summary>Partial log (click to expand)</summary><p>

```
github.com/opentracing/opentracing-go/ext
github.com/ipfs/go-log/tracer/wire
github.com/libp2p/go-libp2p-crypto/pb
github.com/libp2p/go-libp2p-crypto
github.com/ipfs/go-log/tracer
github.com/ipfs/go-log
github.com/libp2p/go-libp2p-peer
github.com/holochain/holochain-proto/peerqueue
builder for '/nix/store/n5vgfhh09kmy70iib97c4057f1h5l49x-holochain-go0.1.0-alpha.drv' failed with exit code 8
error: build of '/nix/store/n5vgfhh09kmy70iib97c4057f1h5l49x-holochain-go0.1.0-alpha.drv' failed
```
</p></details>

zimbatm commented:
@gavinrogers has holochain-go been exclusively developed on a mac? The Linux compilation is failing with an exit status 8.

gavinrogers commented:
@zimbatm No it's been developed on linux machines as well. It's complaining about github.com/gxed/eventfd ... which isn't called by holochain-proto (that I can tell). Is it being called by a dependency? I'll add it

Commit:
Fix: remove platform declaration, add eventfd

"if platforms is not set it will default to go.meta.platforms which is probably
what you want" And that does sound like what we want. Also, the linux build was
complaining about eventfd, or the lack thereof.

gavinrogers commented:
@GrahamcOfBorg build holochain-go

xeji commented:
I'm afraid you lack the access rights to trigger the build.
@GrahamcOfBorg build holochain-go

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42937&attempt_id=4aaf9d8d-0fb7-4990-bc6a-709d9b788da0)

Attempted: holochain-go

<details><summary>Partial log (click to expand)</summary><p>

```
github.com/libp2p/go-libp2p-crypto/pb
github.com/ipfs/go-log/tracer/wire
github.com/multiformats/go-multihash
github.com/libp2p/go-libp2p-crypto
github.com/ipfs/go-log/tracer
github.com/ipfs/go-log
github.com/libp2p/go-libp2p-peer
github.com/holochain/holochain-proto/peerqueue
builder for '/nix/store/ibzp78jm06j5ayfn7zpfga06jc0m1xci-holochain-go0.1.0-alpha.drv' failed with exit code 8
error: build of '/nix/store/ibzp78jm06j5ayfn7zpfga06jc0m1xci-holochain-go0.1.0-alpha.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42937&attempt_id=793f3eea-685b-4234-b289-b9e3fe1a836d)

Attempted: holochain-go

<details><summary>Partial log (click to expand)</summary><p>

```
github.com/libp2p/go-libp2p-crypto/pb
github.com/ipfs/go-log/tracer/wire
github.com/libp2p/go-libp2p-crypto
github.com/ipfs/go-log/tracer
github.com/ipfs/go-log
github.com/libp2p/go-libp2p-peer
github.com/holochain/holochain-proto/hash
github.com/holochain/holochain-proto/peerqueue
builder for '/nix/store/s4s4ca40rbf4fn2hkv647bjpdnip4wmg-holochain-go0.1.0-alpha.drv' failed with exit code 8
error: build of '/nix/store/s4s4ca40rbf4fn2hkv647bjpdnip4wmg-holochain-go0.1.0-alpha.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42937&attempt_id=f85ddc50-e5fc-4cb8-bc86-f750cc6013a1)

Attempted: holochain-go

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/dsdpxy1r5jws4y7wh1csvzr5br4llyz1-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/bin/test.examples: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/bash"
/nix/store/dsdpxy1r5jws4y7wh1csvzr5br4llyz1-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/bs.start: interpreter directive changed from "/bin/sh" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/sh"
/nix/store/dsdpxy1r5jws4y7wh1csvzr5br4llyz1-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/hc.sync: interpreter directive changed from "/bin/sh" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/sh"
/nix/store/dsdpxy1r5jws4y7wh1csvzr5br4llyz1-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/post.up: interpreter directive changed from "/bin/sh" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/sh"
/nix/store/dsdpxy1r5jws4y7wh1csvzr5br4llyz1-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/pre.up: interpreter directive changed from "/bin/sh" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/sh"
/nix/store/dsdpxy1r5jws4y7wh1csvzr5br4llyz1-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/sync: interpreter directive changed from "/bin/sh" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/sh"
/nix/store/dsdpxy1r5jws4y7wh1csvzr5br4llyz1-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/syncCheckFiles: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/bash"
/nix/store/dsdpxy1r5jws4y7wh1csvzr5br4llyz1-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/syncStart: interpreter directive changed from "/bin/sh" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/sh"
/nix/store/dsdpxy1r5jws4y7wh1csvzr5br4llyz1-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/scriptsAccessibleByDockerfiles/entrypoint.sh.addHostUserToContainer: interpreter directive changed from "/bin/sh" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/sh"
/nix/store/54pq676dbg6gnqzp9viskg4lvmxsh7a6-holochain-go0.1.0-alpha-bin
```
</p></details>

gavinrogers commented:
Ok it needs github.com/gxed/GoEndian, but only on linux? I don't fully understand why it's only complaining on some platforms. I should add it, but perhaps I should find out what is using it, and package that separately, with GoEndian as a dependency?

Commit:
Add: GoEndian dependancy

I'm a little confused as to why this buils and runs on darwin, but fails to
build without it on linux. The error was:

```go/src/github.com/gxed/eventfd/eventfd.go:17:2: cannot find package "github.com/gxed/GoEndian" in any of:```

zimbatm commented:
@GrahamcOfBorg build holochain-go

@gavinrogers make sure to use `--option sandbox true` when building on Linux. It's not enabled by default but is required by Hydra to make sure that all the dependencies are declared. It's possible that when testing without the sandboxing the go toolchain will fetch these from the network directly.

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42937&attempt_id=93380a35-5b1d-4736-b875-ba53debf3fc6)

Attempted: holochain-go

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/g03wrz55ddfsm9rg9hg66vpnb1sgnnam-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/bin/holochain.core.fromLocalFilesystem.install: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/8zkg9ac4s4alzyf4a8kfrig1j73z66dw-bash-4.4-p23/bin/bash"
/nix/store/g03wrz55ddfsm9rg9hg66vpnb1sgnnam-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/hc.sync: interpreter directive changed from "/bin/sh" to "/nix/store/8zkg9ac4s4alzyf4a8kfrig1j73z66dw-bash-4.4-p23/bin/sh"
/nix/store/g03wrz55ddfsm9rg9hg66vpnb1sgnnam-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/sync: interpreter directive changed from "/bin/sh" to "/nix/store/8zkg9ac4s4alzyf4a8kfrig1j73z66dw-bash-4.4-p23/bin/sh"
/nix/store/g03wrz55ddfsm9rg9hg66vpnb1sgnnam-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/bs.start: interpreter directive changed from "/bin/sh" to "/nix/store/8zkg9ac4s4alzyf4a8kfrig1j73z66dw-bash-4.4-p23/bin/sh"
/nix/store/g03wrz55ddfsm9rg9hg66vpnb1sgnnam-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/pre.up: interpreter directive changed from "/bin/sh" to "/nix/store/8zkg9ac4s4alzyf4a8kfrig1j73z66dw-bash-4.4-p23/bin/sh"
/nix/store/g03wrz55ddfsm9rg9hg66vpnb1sgnnam-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/syncStart: interpreter directive changed from "/bin/sh" to "/nix/store/8zkg9ac4s4alzyf4a8kfrig1j73z66dw-bash-4.4-p23/bin/sh"
/nix/store/g03wrz55ddfsm9rg9hg66vpnb1sgnnam-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/syncCheckFiles: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/8zkg9ac4s4alzyf4a8kfrig1j73z66dw-bash-4.4-p23/bin/bash"
/nix/store/g03wrz55ddfsm9rg9hg66vpnb1sgnnam-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/post.up: interpreter directive changed from "/bin/sh" to "/nix/store/8zkg9ac4s4alzyf4a8kfrig1j73z66dw-bash-4.4-p23/bin/sh"
checking for references to /build in /nix/store/g03wrz55ddfsm9rg9hg66vpnb1sgnnam-holochain-go0.1.0-alpha...
/nix/store/6hrv1a40ky7a39r3s3v6pkj9qg99ysxq-holochain-go0.1.0-alpha-bin
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42937&attempt_id=d7cda220-fa46-40b7-831c-5c592f3b858a)

Attempted: holochain-go

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/1fnmb8fi55akmkvss707jxb2f5j9dvw2-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/bs.start: interpreter directive changed from "/bin/sh" to "/nix/store/p0vy17dp9jk2mvqsxsqnb14s3797lay7-bash-4.4-p23/bin/sh"
/nix/store/1fnmb8fi55akmkvss707jxb2f5j9dvw2-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/hc.sync: interpreter directive changed from "/bin/sh" to "/nix/store/p0vy17dp9jk2mvqsxsqnb14s3797lay7-bash-4.4-p23/bin/sh"
/nix/store/1fnmb8fi55akmkvss707jxb2f5j9dvw2-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/post.up: interpreter directive changed from "/bin/sh" to "/nix/store/p0vy17dp9jk2mvqsxsqnb14s3797lay7-bash-4.4-p23/bin/sh"
/nix/store/1fnmb8fi55akmkvss707jxb2f5j9dvw2-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/pre.up: interpreter directive changed from "/bin/sh" to "/nix/store/p0vy17dp9jk2mvqsxsqnb14s3797lay7-bash-4.4-p23/bin/sh"
/nix/store/1fnmb8fi55akmkvss707jxb2f5j9dvw2-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/sync: interpreter directive changed from "/bin/sh" to "/nix/store/p0vy17dp9jk2mvqsxsqnb14s3797lay7-bash-4.4-p23/bin/sh"
/nix/store/1fnmb8fi55akmkvss707jxb2f5j9dvw2-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/syncCheckFiles: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/p0vy17dp9jk2mvqsxsqnb14s3797lay7-bash-4.4-p23/bin/bash"
/nix/store/1fnmb8fi55akmkvss707jxb2f5j9dvw2-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/syncStart: interpreter directive changed from "/bin/sh" to "/nix/store/p0vy17dp9jk2mvqsxsqnb14s3797lay7-bash-4.4-p23/bin/sh"
/nix/store/1fnmb8fi55akmkvss707jxb2f5j9dvw2-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/scriptsAccessibleByDockerfiles/entrypoint.sh.addHostUserToContainer: interpreter directive changed from "/bin/sh" to "/nix/store/p0vy17dp9jk2mvqsxsqnb14s3797lay7-bash-4.4-p23/bin/sh"
checking for references to /build in /nix/store/1fnmb8fi55akmkvss707jxb2f5j9dvw2-holochain-go0.1.0-alpha...
/nix/store/z9lsjgjac5cr59wcykkzrmgwb4kc84w9-holochain-go0.1.0-alpha-bin
```
</p></details>

zimbatm commented:
Also I recommend vendoring the dependencies using a tool like [godep](https://github.com/tools/godep). This will simplify the nix packaging as the deps.nix doesn't need to be generated anymore, and also allow anyone with go installed to easily run `go get github.com/holochain/holochain-proto` to build their own binary version.

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.42937&attempt_id=e5b6c442-4e0d-4125-b82c-954b9b46c01a)

Attempted: holochain-go

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/lw6b4d9ap5r53xai87w871n0wgwr4y4q-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/bin/test.examples: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/bash"
/nix/store/lw6b4d9ap5r53xai87w871n0wgwr4y4q-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/bs.start: interpreter directive changed from "/bin/sh" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/sh"
/nix/store/lw6b4d9ap5r53xai87w871n0wgwr4y4q-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/hc.sync: interpreter directive changed from "/bin/sh" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/sh"
/nix/store/lw6b4d9ap5r53xai87w871n0wgwr4y4q-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/post.up: interpreter directive changed from "/bin/sh" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/sh"
/nix/store/lw6b4d9ap5r53xai87w871n0wgwr4y4q-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/pre.up: interpreter directive changed from "/bin/sh" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/sh"
/nix/store/lw6b4d9ap5r53xai87w871n0wgwr4y4q-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/sync: interpreter directive changed from "/bin/sh" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/sh"
/nix/store/lw6b4d9ap5r53xai87w871n0wgwr4y4q-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/syncCheckFiles: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/bash"
/nix/store/lw6b4d9ap5r53xai87w871n0wgwr4y4q-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/docker/app.docker/syncedMultiNode.containers.scripts/syncStart: interpreter directive changed from "/bin/sh" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/sh"
/nix/store/lw6b4d9ap5r53xai87w871n0wgwr4y4q-holochain-go0.1.0-alpha/share/go/src/github.com/holochain/holochain-proto/scriptsAccessibleByDockerfiles/entrypoint.sh.addHostUserToContainer: interpreter directive changed from "/bin/sh" to "/nix/store/q2wqq1k20v8kc3vckapqf5nws30brnni-bash-4.4-p23/bin/sh"
/nix/store/i9n2ar4n5d77d22nwmlw43dc6fzr90mn-holochain-go0.1.0-alpha-bin
```
</p></details>

gavinrogers commented:
@zimbatm "Please use dep or another tool instead." in the README

gavinrogers commented:
@grahamc @zimbatm @svanderburg @edolstra could I get a merge on this? we have a hackathon coming up this weekend and it would be a good demonstration of nixpkgs if participants could install nix, add the unstable channel and just ```nix-shell -p holochain-go```
